packages/block-library/src/button/save.js

```diff
@@ -61,19 +63,28 @@ export default function save( { attributes, className } ) {
 		[ `has-custom-width wp-block-button__width-${ width }` ]: width,
 		[ `has-custom-font-size` ]: fontSize || style?.typography?.fontSize,
 	} );
-
 	return (
 		<div { ...useBlockProps.save( { className: wrapperClasses } ) }>
-			<RichText.Content
-				tagName="a"
-				className={ buttonClasses }
-				href={ url }
-				title={ title }
-				style={ buttonStyle }
-				value={ text }
-				target={ linkTarget }
-				rel={ rel }
-			/>
+			{ 'button' === TagName ? (
+				<button
```

Comment:
Whats the reason for this conditional here? Couldn't this also be handled by the `RichText.Content` with the `tagName` set to `button`? 

And if not would it make sense to call out why in a comment? :)

Reply:
@aristath Even with the comment I'm still not entirely sure on the approach. 

I think maybe something like this would make it more clear:

```js

const buttonElementProps = {
    // sharedProps
};

if ( 'button' === TagName ) {
    buttonElementProps.attribute = 'something';
}

return (
    <RichText.Content {... buttonElementProps} />
);
```

Reply:
Or if not that the `button` element should also use the `RichText.Content` since it was created by a `RichText` element.

Reply:
Updated the implementation 👍